eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml

```diff
@@ -121,16 +121,40 @@
     <!-- Known issue in autorest where it will generate a variable with a value of null then use it in a not null check.
     https://github.com/Azure/autorest.java/issues/328 -->
     <Match>
-        <Class name="~com\.microsoft\.azure\.batch\.protocol\.implementation\.[\w]+Impl"/>
-        <Bug pattern="RCN_REDUNDANT_NULLCHECK_OF_NULL_VALUE"/>
+      <Class name="~com\.microsoft\.azure\.batch\.protocol\.implementation\.[\w]+Impl"/>
+      <Bug pattern="RCN_REDUNDANT_NULLCHECK_OF_NULL_VALUE"/>
     </Match>
 
     <Match>
-        <Class name="com.microsoft.azure.batch.CreateTasksErrorException"/>
-        <Or>
-            <Field name="failureTaskList"/>
-            <Field name="pendingTaskList"/>
-        </Or>
-        <Bug pattern="SE_BAD_FIELD"/>
+      <Class name="com.microsoft.azure.batch.CreateTasksErrorException"/>
+      <Or>
+        <Field name="failureTaskList"/>
+        <Field name="pendingTaskList"/>
+      </Or>
+      <Bug pattern="SE_BAD_FIELD"/>
+    </Match>
+
+    <!-- Methods in Common package return null instead of meaningful instances when resourses do not exist. Will address the bugs in Track Two.
+    -->
+    <Match>
+      <Class name="~com\.azure\.common\.implementation\.(Base64Url|serializer\.HttpResponseDecodeData|util\.CloneUtils)"/>
+      <Bug pattern="PZLA_PREFER_ZERO_LENGTH_ARRAYS"/>
+    </Match>
+    <Match>
+      <Class name="com.azure.common.implementation.Base64Url"/>
+      <Bug pattern="NP_TOSTRING_COULD_RETURN_NULL"/>
+    </Match>
+
+    <!-- There is no need to make serializer or deserializer classes serializable. Suppress bugs related to these classes.
+    -->
+    <Match>
+      <Class name="~com\.azure\.common\.implementation\.serializer\.jackson\.(AdditionalPropertiesDeserializer|AdditionalPropertiesSerializer|FlatteningDeserializer|FlatteningSerializer)"/>
+      <Bug pattern="SE_NO_SERIALVERSIONID"/>
+    </Match>
```

Comment:
There would be no harm in adding a serial version ID here, would there // cc @anuchandy

Reply:
I have discussed with @anuchandy . If add version id, it will complain the field not serializable. When add keyword transient, the spotbugs will detect the field being using many places bug. So we decided to suppress the serializable bug check as the class does not intent to be serializable.

Reply:
I'm fine with that. More generally, I wonder if we just disable this check in SpotBugs.

Reply:
I did not see the point of making serializer and deserializer serializable.